sklearn/tree/tree.py

```diff
@@ -309,6 +309,27 @@ def fit(self, X, y, sample_weight=None, check_input=True):
 
         return self
 
+    def _validate_X_predict(self, X, check_input):
+        """Validate X whenever one tries to predict, apply, predict_proba"""
+        if self.tree_ is None:
+            raise NotFittedError("Tree not initialized. Perform a fit first")
```

Comment:
Not really important, but it would be nice to use the same sentence in forests and trees upon `NotFittedError`